### ISSUE TITLE
Positive deltas are not accepted in many character sheet fields

### ISSUE THREAD

**Issue by benbarbour** [foundryvtt/dnd5e#1806]
**Foundry Version:** 10.285
**DnD5e Version:** 2.0.2
**Modules:** None

If I enter "+X" in a currency box on the inventory panel of the Actor sheet then that currency is set to X. I expected X to be added to the current value. If I enter "-X" instead then X is subtracted from the current value. Using no prefix sets the value to whatever I entered, as expected.

*UPDATE*: This affects more than just the currency fields (see below).

**Comment by unsoluble** [comment]
Note this is also the case for other places where deltas should be accepted, like XP and HP.

**Comment by kaelad02** [comment]
I remember something about some `<input>` fields changing from input type string to number. Could that be it? Could the fields now only be accepting numbers, so when they're submitted they're changed from `+10` as a string to just `10` the number?

**Comment by LiamFallon213** [comment]
So I did some testing. What I found, all testing done on the currency fields.

- This bug only happens when you add a positive delta to a non zero field.

- If you manually go into the HTML of foundry and change the input of the file to text instead of number. The bug is not present.

- If you manually set the currency as text before run, neither deltas work.

- The code for handling delta's simply is not running when the bug is present.

- The code for activating listeners, (around line 600 in base-sheet.mjs) does not run when bug is present.

This is where my investigation into the bug currently is stuck at, I can't follow the data up past the listener activation failure. I will provide more info when I discover it.

**Comment by arbron** [comment]
Looks like this is only an issue in Chrome, it works fine in Firefox. It seems like Chrome is removing the leading plus from the value before we get a chance to handle it (e.g. `"+15"` turns into `"15"`) while Firefox leaves it alone.

**Comment by LiamFallon213** [comment]
> Looks like this is only an issue in Chrome, it works fine in Firefox. It seems like Chrome is removing the leading plus from the value before we get a chance to handle it (e.g. `"+15"` turns into `"15"`) while Firefox leaves it alone.

I can confirm this, I was able to get a proof of concept working where I changed the input type to text. Then changed the delta function to handle this by parsing to float in the addition line, ~line 700 base-sheet.mjs.
`input.value = parseFloat(foundry.utils.getProperty(this.actor, input.name)) + delta;`

Notably this breaks a bunch of other things, so I can't just do this then push a fix. Specifically it's breaking the CSS right now, which I'm working on.
